siliconcompiler/scheduler.py

```diff
@@ -3,68 +3,71 @@
 import subprocess
 import stat
 import time
+import uuid
+import json
 
 ###########################################################################
 def _deferstep(chip, step, index, status):
     '''
     Helper method to run an individual step on a slurm cluster.
-    If a base64-encoded 'decrypt_key' is set in the Chip's status
-    dictionary, the job's data is assumed to be encrypted,
-    and a more complex command is assembled to ensure that the data
-    is only decrypted temporarily in the compute node's local storage.
     '''
 
     # Determine which HPC job scheduler being used.
     scheduler_type = chip.get('option', 'scheduler', 'name', step=step, index=index)
 
+    if scheduler_type != 'slurm':
+        raise ValueError(f'{scheduler_type} is not a supported scheduler')
+
     # Determine which cluster parititon to use. (Default value can be overridden on per-step basis)
     partition = chip.get('option', 'scheduler', 'queue', step=step, index=index)
+    if not partition:
+        partition = _get_slurm_partitions()
+
     # Get the temporary UID associated with this job run.
-    job_hash = chip.status['jobhash']
+    if 'jobhash' in chip.status:
+        job_hash = chip.status['jobhash']
+    else:
+        # Generate a new uuid since it was not set
+        job_hash = uuid.uuid4().hex
+
+    job_name = f'{job_hash}_{step}{index}'
 
-    # Job data is not encrypted, so it can be run in shared storage.
     # Write out the current schema for the compute node to pick up.
     job_dir = chip._getworkdir()
     cfg_dir = f'{job_dir}/configs'
     cfg_file = f'{cfg_dir}/{step}{index}.json'
     if not os.path.isdir(cfg_dir):
         os.mkdir(cfg_dir)
-    chip.unset('option', 'scheduler', 'name', step=step, index=index)
+    chip.set('option', 'scheduler', 'name', None, step=step, index=index)
```

Comment:
Is `unset` unsuitable here? Does it fail to unset a globally-defined value when step/index are specified?

Reply:
Yes, the `.unset` unfortunately does not behave correctly (that was a fun loop to hunt for).
@nmoroze is going to fix it, but where we are using it, it's likely not doing what we want.

Reply:
Okay - maybe comment out the `unset` with a TODO, if we intend to fix it?